service_management/azure/lib/azure/service_bus/auth/shared_access_signer.rb

```diff
@@ -23,9 +23,9 @@ def name
         end
 
         def token(uri)
-          url_encoded_resource = CGI.escape(uri.to_s.downcase).gsub('+', '%20').downcase
+          url_encoded_resource = URI.escape(uri.to_s.downcase).downcase
```

Comment:
I was reading about [the difference between uri escape and cgi escape](http://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape), have you looked into this? Maybe we should not be using URI.escape?